server/channels/jobs/delete_empty_drafts_migration/delete_empty_drafts_migration.go

```diff
@@ -0,0 +1,224 @@
+	GetClusterStatus() []*model.ClusterInfo
+}
+
+type DeleteEmptyDraftsMigrationWorker struct {
+	name      string
+	jobServer *jobs.JobServer
+	store     store.Store
+	app       AppIFace
+
+	stop    chan bool
+	stopped chan bool
+	jobs    chan model.Job
+}
+
+func MakeWorker(jobServer *jobs.JobServer, store store.Store, app AppIFace) model.Worker {
+	worker := &DeleteEmptyDraftsMigrationWorker{
+		jobServer: jobServer,
+		store:     store,
+		app:       app,
+		name:      JobName,
+		stop:      make(chan bool, 1),
+		stopped:   make(chan bool, 1),
+		jobs:      make(chan model.Job),
+	}
+	return worker
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) Run() {
+	mlog.Debug("Worker started", mlog.String("worker", worker.name))
+	// We have to re-assign the stop channel again, because
+	// it might happen that the job was restarted due to a config change.
+	worker.stop = make(chan bool, 1)
+
+	defer func() {
+		mlog.Debug("Worker finished", mlog.String("worker", worker.name))
+		worker.stopped <- true
+	}()
+
+	for {
+		select {
+		case <-worker.stop:
+			mlog.Debug("Worker received stop signal", mlog.String("worker", worker.name))
+			return
+		case job := <-worker.jobs:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", worker.name))
+			worker.DoJob(&job)
+		}
+	}
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", worker.name))
+	close(worker.stop)
+	<-worker.stopped
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) JobChannel() chan<- model.Job {
+	return worker.jobs
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) IsEnabled(_ *model.Config) bool {
+	return true
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) getJobMetadata(job *model.Job, key string) (int64, *model.AppError) {
+	countStr := job.Data[key]
+	count := int64(0)
+	var err error
+	if countStr != "" {
+		count, err = strconv.ParseInt(countStr, 10, 64)
+		if err != nil {
+			return 0, model.NewAppError("getJobMetadata", model.NoTranslation, nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+	return count, nil
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) DoJob(job *model.Job) {
+	defer worker.jobServer.HandleJobPanic(job)
+
+	if claimed, err := worker.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("DeleteEmptyDraftsMigrationWorker experienced an error while trying to claim job",
+			mlog.String("worker", worker.name),
+			mlog.String("job_id", job.Id),
+			mlog.Err(err))
+		return
+	} else if !claimed {
+		return
+	}
+
+	var appErr *model.AppError
+	// We get the job again because ClaimJob changes the job status.
+	job, appErr = worker.jobServer.GetJob(job.Id)
+	if appErr != nil {
+		mlog.Error("DeleteEmptyDraftsMigrationWorker: job execution error", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.Err(appErr))
+		worker.setJobError(job, appErr)
+		return
+	}
+
+	// Wait for all clusters to finish DB migration
```

Comment:
Do we need to handle the possibility that an older node starts after the job and starts creating new draft entries? Or even if the cluster is downgraded to recover from an issue and later re-upgraded?